### ISSUE TITLE
CountVar help message is confusing and/or incorrect

### ISSUE THREAD

**Issue by greatroar** [spf13/pflag#267]
CountVar's help message fails to communicate that its argument needs to be given to it using an equals sign.

Namely, if I compile the following as pflagtest, with either v1.0.5 or 81378bbcd8a1005f72b1e8d7579e5dd7b2d612ab,

```go
package main

import (
	"fmt"
	"github.com/spf13/pflag"
)

func main() {
	var n int
	pflag.CountVar(&n, "foo", "")
	pflag.Parse()

	fmt.Printf("foo count: %d\n", n)

	fmt.Println("non-flag arguments:")
	for _, arg := range pflag.Args() {
		fmt.Printf("    %q\n", arg)
	}
}
```

then I get the following behavior:

```
# ok
$ ./pflagtest --foo=2
foo count: 2
non-flag arguments:

# help says --foo 2 should work too:
$ ./pflagtest --help
Usage of ./pflagtest:
      --foo count   
pflag: help requested

# but it doesn't
$ ./pflagtest --foo 2
foo count: 1
non-flag arguments:
    "2"

```

Spotted in restic/restic#2805.

**Comment by cornfeedhobo** [comment]
@greatroar count accepts `=` or the additive form. Any ideas on how to represent both?

**Comment by greatroar** [comment]
Do you mean `--foo --foo`?

**Comment by cornfeedhobo** [comment]
@greatroar correct. If you look at [the tests](https://github.com/spf13/pflag/blob/master/count_test.go#L22-L25) you can see some examples of the various "additive" forms.

**Comment by tomasaschan** [comment]
This issue has not been updated in a long time. Is it still relevant?

(If I receive no answer before the end of the summer, I'll close it. Feel free to re-open if you come here after that happens and it is relevant to you - but if you do, please also comment and specify what outcome you're interested in!)
